### PR TITLE
gms,service: add a feature to protect the usage of allow_mutation_read_page_without_live_row

### PR DESCRIPTION
**Pull request by denesb**
allow_mutation_read_page_without_live_row is a new option in the partition_slice::option option set. In a mixed clusters, old nodes possibly don't know this new option, so its usage must be protected by a cluster feature. This patch does just that.

Fixes: #15795

### CODE REVIEW

**Comment by avikivity** [comment]
Aha, I remember that we sent the new bit without a feature thinking it will be ignored, but it's caught.

**Comment by denesb** [comment]
I don't understand all this failures on such a simple patch. Rebased.

**Comment by avikivity** [comment]
> I don't understand all this failures on such a simple patch. Rebased.

Ran on m5ad.8xlarge, so should be okay disk-wise.

**Comment by avikivity** [comment]
Actually it ran on many instance types, but all have nvme as far as I can tell.

**Comment by avikivity** [comment]
@fruch is there any way to tell if the mount procedure worked correctly?

**Comment by fruch** [comment]
Now, we are not collecting logs from that stage.

Anyhow those failures are too reproducible, it doesn't fit the patterns we've seen when disk wasn't configured

**Comment by avikivity** [comment]
yes, looks like a genuine bug uncovered by CI (like it's supposed to)

**Comment by denesb** [comment]
Looks like a genuine bug related to view schemas. Nodes are crashing due to uninitialized schema: https://github.com/scylladb/scylladb/issues/15904

**Comment by denesb** [comment]
All the failing tests I checked so far are related to schema problems.

**Comment by denesb** [comment]
DO NOT MERGE!!!!

I pushed a version which has the if condition commented, to see if that is what triggers all these failures. The patch now effectively just adds a feature flag.

**Comment by kbr-scylla** [comment]
I looked at the test_topology_ops failure and created https://github.com/scylladb/scylladb/issues/15935.

There are a couple of bugs that have already been detected by the test. For now I'll disable the part of the test which is detecting these issues (the one which is performing CQL writes) until we fix all known bugs.

**Comment by denesb** [comment]
> I looked at the test_topology_ops failure and created #15935.
> 
> There are a couple of bugs that have already been detected by the test. For now I'll disable the part of the test which is detecting these issues (the one which is performing CQL writes) until we fix all known bugs.

Thanks. Why is this particular PR triggering this so reliably? I also saw this test fail at other occasions, but here it is very consistent.

**Comment by denesb** [comment]
> Thanks. Why is this particular PR triggering this so reliably? I also saw this test fail at other occasions, but here it is very consistent.

Actually, it is not that consistent.

**Comment by denesb** [comment]
I have no other choice but to dig into these failures. Seems like there is a real problem behind all this.

**Comment by denesb** [comment]
BTW, with this PR, there are now in total 106 tests.
With a ScyllaDB dev build, it takes  21.65s to run all these in a sequential manner.
The same takes 96.26s to run the same with the Java nodetool (which even skips 12 tests).

So even with all the seastar machinery, which is massive overkill for a simple tool such as nodetool, the scylla-native nodetool is still > 5X faster than the Java-based one. Looks like the JVM is even more overkill for a simple tool, than seastar.

**Comment by denesb** [comment]
> [test_simple_kill_node_while_decommissioning](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/4504/testReport/junit/update_cluster_layout_tests/TestUpdateClusterLayout/Tests___Sanity_Tests___test_simple_kill_node_while_decommissioning) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_simple_kill_node_while_decommissioning)

https://github.com/scylladb/scylladb/issues/15146

**Comment by denesb** [comment]
>[topology_experimental_raft.test_topology_ops.dev.2](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/4504/testReport/junit/%28root%29/non-boost%20tests/Tests___Unit_Tests___topology_experimental_raft_test_topology_ops_dev_2) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+topology_experimental_raft.test_topology_ops.dev.2)
[topology_experimental_raft.test_topology_ops.dev.3](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/4504/testReport/junit/%28root%29/non-boost%20tests/Tests___Unit_Tests___topology_experimental_raft_test_topology_ops_dev_3) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+topology_experimental_raft.test_topology_ops.dev.3)

https://github.com/scylladb/scylladb/issues/15924

**Comment by kbr-scylla** [comment]
> https://github.com/scylladb/scylladb/issues/15924

Rebase -- these background writes are disabled on `master` (6cc5bcae801d2e1298870acd3b3432fbb6aa3c57)

**Comment by denesb** [comment]
> [test_off_strategy_during_bootstrap](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/4504/testReport/junit/bootstrap_test/TestBootstrap/Tests___Sanity_Tests___test_off_strategy_during_bootstrap) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_off_strategy_during_bootstrap)

https://github.com/scylladb/scylladb/issues/16004

**Comment by denesb** [comment]
[test_cluster_cleanup](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/4470/testReport/junit/cleanup_test/TestCleanup/Tests___Sanity_Tests___test_cluster_cleanup) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_cluster_cleanup)

https://github.com/scylladb/scylladb/issues/16004

**Comment by denesb** [comment]
> [test_tracing_shutdown](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/4504/testReport/junit/cql_tracing_test/TestCqlTracing/Tests___Sanity_Tests___test_tracing_shutdown) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_tracing_shutdown)

https://github.com/scylladb/scylladb/issues/16004 with a twist. Writes to `system_traces` fail to lookup the table by version (due to different version), so they call back to the originating node, which is shutting down and has already unregistered its RPC verbs. Not sure if this deserves a separate bug or not.

**Comment by denesb** [comment]
Looks like almost all failed tests are related to https://github.com/scylladb/scylladb/issues/16004.

**Comment by denesb** [comment]
> > #15924
> 
> Rebase -- these background writes are disabled on `master` ([6cc5bca](https://github.com/scylladb/scylladb/commit/6cc5bcae801d2e1298870acd3b3432fbb6aa3c57))

Rebased.

**Comment by denesb** [comment]
Rebased on top of https://github.com/scylladb/scylladb/commit/22381441b00eb127c7ca656ae928a1567bd964f0 (the fix for https://github.com/scylladb/scylladb/issues/16004).

**Comment by bhalevy** [comment]
> Rebased on top of [2238144](https://github.com/scylladb/scylladb/commit/22381441b00eb127c7ca656ae928a1567bd964f0) (the fix for #16004).

Cc @mykaul

**Comment by bhalevy** [comment]
@kbr-scylla / @avikivity do you need to re-review or can this PR be merged as is?

**Review by kbr-scylla — APPROVED** [review]
I'd like at least one other person who knows this code, preferably someone who reviewed the previous PR (which extended the enum_set), to review this one before I merge.

**Comment by denesb** [comment]
> I'd like at least one other person who knows this code, preferably someone who reviewed the previous PR (which extended the enum_set), to review this one before I merge.

That person is @tgrabiec.

**Review by tgrabiec — APPROVED** [review]

**Pull request closed without merging**
